src/univariate/discrete/categorical.jl

```diff
@@ -80,7 +80,10 @@ function cdf(d::Categorical{T}, x::Int) where T<:Real
     return c
 end
 
-pdf(d::Categorical{T}, x::Int) where {T<:Real} = insupport(d, x) ? probs(d)[x] : zero(T)
+function pdf(d::Categorical, x::Real)
+    ps = probs(d)
+    return insupport(d, x) ? ps[round(Int, x)] : zero(eltype(ps))
```

Comment:
This should be convert?

Reply:
It's not completely consistent throughout the package, but I got the impression that (more) often `round` is used in Distributions for such cases (https://github.com/JuliaStats/Distributions.jl/search?q=round%28Int).